### ISSUE TITLE
Cockroach failed to start after cold boot

### ISSUE THREAD

**Issue by jgallagher** [oxidecomputer/omicron#4087]
After a mupdate + reboot of the dogfood rack, one of the CRDB zones showed that CRDB failed to start:

```
BRM42220017 # zlogin oxz_cockroachdb_3237a532-acaa-4ebe-bf11-dde794fea739
root@oxz_cockroachdb_3237a532-acaa-4ebe-bf11-dde794fea739:~# svcs -xv
svc:/oxide/cockroachdb:default (CockroachDB)
 State: maintenance since Tue Sep 12 20:40:47 2023
Reason: Restarting too quickly.
   See: http://illumos.org/msg/SMF-8000-L5
   See: /var/svc/log/oxide-cockroachdb:default.log
Impact: This service is not running.
```

Logfiles have been copied to `/staff/core/omicron-4087`.

In the `svcs` log `rotated-zone-logs/3cafbb47-c194-4a42-99ff-34dfeab999ed/crypt/debug/oxz_cockroachdb_3237a532-acaa-4ebe-bf11-dde794fea739/oxide-cockroachdb:default.log.1694551247`, we see that we tried to start CRDB several times:

```
/staff/core/omicron-4087 $ grep ' Sep ' rotated-zone-logs/3cafbb47-c194-4a42-99ff-34dfeab999ed/crypt/debug/oxz_cockroachdb_3237a532-acaa-4ebe-bf11-dde794fea739/oxide-cockroachdb\:default.log.1694551247
[ Sep 12 20:40:12 Enabled. ]
[ Sep 12 20:40:12 Rereading configuration. ]
[ Sep 12 20:40:13 Rereading configuration. ]
[ Sep 12 20:40:24 Executing start method ("/opt/oxide/lib/svc/manifest/cockroachdb.sh"). ]
[ Sep 12 20:40:26 Method "start" exited with status 0. ]
[ Sep 12 20:40:28 Stopping because all processes in service exited. ]
[ Sep 12 20:40:28 Executing stop method (:kill). ]
[ Sep 12 20:40:28 Executing start method ("/opt/oxide/lib/svc/manifest/cockroachdb.sh"). ]
[ Sep 12 20:40:34 Method "start" exited with status 0. ]
[ Sep 12 20:40:36 Stopping because all processes in service exited. ]
[ Sep 12 20:40:36 Executing stop method (:kill). ]
[ Sep 12 20:40:36 Executing start method ("/opt/oxide/lib/svc/manifest/cockroachdb.sh"). ]
[ Sep 12 20:40:38 Method "start" exited with status 0. ]
[ Sep 12 20:40:39 Stopping because all processes in service exited. ]
[ Sep 12 20:40:39 Executing stop method (:kill). ]
[ Sep 12 20:40:39 Executing start method ("/opt/oxide/lib/svc/manifest/cockroachdb.sh"). ]
[ Sep 12 20:40:45 Method "start" exited with status 0. ]
[ Sep 12 20:40:47 Stopping because all processes in service exited. ]
[ Sep 12 20:40:47 Executing stop method (:kill). ]
[ Sep 12 20:40:47 Restarting too quickly, changing state to maintenance. ]
```

Each attempt failed with a similar (uninformative) message:

```
*
* WARNING: ALL SECURITY CONTROLS HAVE BEEN DISABLED!
*
* This mode is intended for non-production testing only.
*
* In this mode:
* - Your cluster is open to any client that can access fd00:1122:3344:109::3.
* - Intruders with access to your machine or network can observe client-server traffic.
* - Intruders can log in without password and read or write any data in the cluster.
* - Intruders can consume all your server's resources and cause unavailability.
*
*
* INFO: To start a secure server without mandating TLS for clients,
* consider --accept-sql-without-tls instead. For other options, see:
*
* - https://go.crdb.dev/issue-v/53404/v22.1
* - https://www.cockroachlabs.com/docs/v22.1/secure-a-cluster.html
*
*
* ERROR: Queued as error e10204b4bdd3470d98244a999e1d3327
*
```

The cockroachdb logs themselves are more revealing. From `cockroach-db-logs/cockroach.oxzcockroachdb3237a532-acaa-4ebe-bf11-dde794fea739.root.2023-09-12T20_40_45Z.010070.log`:

```
I230912 20:40:46.181815 153 1@server/server.go:1454 ⋮ [n1] 37  advertising CockroachDB node at ‹[fd00:1122:3344:109::3]:32221›
E230912 20:40:46.181843 153 server/status/recorder.go:420 ⋮ [n1,summaries] 38  could not get total system memory: ‹not implemented on illumos›
I230912 20:40:46.182600 854 1@vendor/github.com/cockroachdb/circuitbreaker/circuitbreaker.go:322 ⋮ [n1] 39  circuitbreaker: ‹rpc [fd00:1122:3344:109::3]:32221 [n2]› tripped: unable to look up descriptor for n2
I230912 20:40:46.182728 854 1@vendor/github.com/cockroachdb/circuitbreaker/circuitbreaker.go:447 ⋮ [n1] 40  circuitbreaker: ‹rpc [fd00:1122:3344:109::3]:32221 [n2]› event: ‹BreakerTripped›
I230912 20:40:46.182600 851 1@vendor/github.com/cockroachdb/circuitbreaker/circuitbreaker.go:322 ⋮ [n1] 41  circuitbreaker: ‹rpc [fd00:1122:3344:109::3]:32221 [n4]› tripped: unable to look up descriptor for n4
I230912 20:40:46.182765 851 1@vendor/github.com/cockroachdb/circuitbreaker/circuitbreaker.go:447 ⋮ [n1] 42  circuitbreaker: ‹rpc [fd00:1122:3344:109::3]:32221 [n4]› event: ‹BreakerTripped›
I230912 20:40:46.182646 852 1@vendor/github.com/cockroachdb/circuitbreaker/circuitbreaker.go:322 ⋮ [n1] 43  circuitbreaker: ‹rpc [fd00:1122:3344:109::3]:32221 [n5]› tripped: unable to look up descriptor for n5
I230912 20:40:46.182786 852 1@vendor/github.com/cockroachdb/circuitbreaker/circuitbreaker.go:447 ⋮ [n1] 44  circuitbreaker: ‹rpc [fd00:1122:3344:109::3]:32221 [n5]› event: ‹BreakerTripped›
I230912 20:40:46.182663 853 1@vendor/github.com/cockroachdb/circuitbreaker/circuitbreaker.go:322 ⋮ [n1] 45  circuitbreaker: ‹rpc [fd00:1122:3344:109::3]:32221 [n3]› tripped: unable to look up descriptor for n3
I230912 20:40:46.182804 853 1@vendor/github.com/cockroachdb/circuitbreaker/circuitbreaker.go:447 ⋮ [n1] 46  circuitbreaker: ‹rpc [fd00:1122:3344:109::3]:32221 [n3]› event: ‹BreakerTripped›
I230912 20:40:46.339316 857 1@vendor/github.com/cockroachdb/circuitbreaker/circuitbreaker.go:322 ⋮ [n1] 47  circuitbreaker: ‹rpc [fd00:1122:3344:109::3]:32221 [n5]› tripped: unable to look up descriptor for n5
I230912 20:40:46.339464 857 1@vendor/github.com/cockroachdb/circuitbreaker/circuitbreaker.go:447 ⋮ [n1] 48  circuitbreaker: ‹rpc [fd00:1122:3344:109::3]:32221 [n5]› event: ‹BreakerTripped›
E230912 20:40:47.164939 654 util/log/logcrash/crash_reporting.go:359 ⋮ [n1] 49  Queued as error ‹e10204b4bdd3470d98244a999e1d3327›
I230912 20:40:47.183537 651 1@vendor/github.com/cockroachdb/circuitbreaker/circuitbreaker.go:322 ⋮ [n1] 51  circuitbreaker: ‹gossip [fd00:1122:3344:109::3]:32221->[fd00:1122:3344:107::3]:32221› tripped: stopped
I230912 20:40:47.191460 868 kv/kvserver/liveness/liveness.go:765 ⋮ [n1,liveness-hb] 52  unable to get liveness record from KV: unable to get liveness: aborted in DistSender: context canceled
I230912 20:40:47.191291 651 1@vendor/github.com/cockroachdb/circuitbreaker/circuitbreaker.go:447 ⋮ [n1] 53  circuitbreaker: ‹gossip [fd00:1122:3344:109::3]:32221->[fd00:1122:3344:107::3]:32221› event: ‹BreakerTripped›
W230912 20:40:47.191556 651 gossip/client.go:118 ⋮ [n1] 54  failed to start gossip client to ‹[fd00:1122:3344:107::3]:32221›: stopped
W230912 20:40:47.191856 48 kv/txn.go:728 ⋮ [intExec=‹protectedts-GetMetadata›] 55  failure aborting transaction: ‹node unavailable; try another peer›; abort caused by: aborted in DistSender: context canceled
W230912 20:40:47.192108 47 kv/txn.go:728 ⋮ [-] 56  failure aborting transaction: ‹node unavailable; try another peer›; abort caused by: failed to fetch protectedts metadata: failed to read metadata: ‹protectedts-GetMetadata›: context canceled
F230912 20:40:47.172448 654 1@server/server.go:246 ⋮ [n1] 50  clock synchronization error: this node is more than 500ms away from at least half of the known nodes (1 of 2 are within the offset)
```

followed by the backtraces of hundreds of goroutines. @iliana noted that the timestamps in the above output are out of order; unclear if that means the system clock went backwards or if it's a log printing issue.

I also copyed the `chronyd` logs from the NTP zone. It does look there are some large errors shortly after the cockroachdb failures:

```
===================================================================================================================================
   Date (UTC) Time     IP Address   St   Freq ppm   Skew ppm     Offset L Co  Offset sd Rem. corr. Root delay Root disp. Max. error
===================================================================================================================================
2023-09-12 20:40:03 fd00:1122:3344:104::e 11    -53.326      4.515 -8.966e-06 N  1  5.721e-06  3.737e-01  1.934e-04  1.138e-05  3.765e-01
2023-09-12 20:40:04 fd00:1122:3344:104::e 11    -54.117      4.090 -4.592e-06 N  1  5.717e-06  3.709e-01  1.934e-04  1.373e-05  3.738e-01
2023-09-12 20:40:05 fd00:1122:3344:104::e 11    -54.690      3.398 -4.895e-06 N  1  5.250e-06  3.682e-01  1.939e-04  8.103e-06  3.711e-01
2023-09-12 20:40:07 fd00:1122:3344:104::e 11    -54.455      2.931  3.916e-06 N  1  5.798e-06  3.628e-01  1.939e-04  1.418e-05  3.684e-01
2023-09-12 20:40:08 0.0.0.0          0    -54.455      2.931  0.000e+00 ?  0  0.000e+00  3.609e-01  1.939e-04  1.701e-05  3.629e-01
2023-09-12 20:42:16 fd00:1122:3344:102::e 11    -54.455      2.931  7.329e-01 N  1  1.164e-01  1.421e-02  2.319e-04  1.267e-01  3.615e-01
2023-09-12 20:42:16 fd00:1122:3344:102::e 11    -54.455      2.931 -2.286e-01 N  1  6.688e-02 -7.159e-01  1.432e-04  7.104e-02  8.484e-01
2023-09-12 20:42:17 fd00:1122:3344:102::e 11    -54.455      2.931 -4.391e-02 N  1  3.282e-02 -4.845e-01  1.115e-04  3.808e-02  5.890e-01
2023-09-12 20:42:19 fd00:1122:3344:102::e 11    -54.455      2.931 -2.434e-02 N  1  2.406e-02 -4.373e-01  1.026e-04  4.596e-02  5.189e-01
2023-09-12 20:42:20 fd00:1122:3344:102::e 11    -54.455      2.931 -5.830e-03 N  1  2.560e-02 -4.102e-01  1.110e-04  5.038e-02  4.954e-01
2023-09-12 20:42:21 fd00:1122:3344:102::e 11    -54.451      3.555  3.734e-02 N  1  2.271e-04 -4.016e-01  1.273e-04  2.542e-04  4.847e-01
2023-09-12 20:42:22 fd00:1122:3344:102::e 11    -40.635     29.256 -1.012e-05 N  1  3.684e-06 -4.362e-01  1.170e-04  1.558e-05  4.394e-01
2023-09-12 20:42:23 fd00:1122:3344:102::e 11     -2.292     18.847  3.568e-05 N  1  5.684e-06 -4.335e-01  1.359e-04  1.577e-05  4.363e-01
```

~~Unfortunately the sled-agent log (same directory) appears to be cut off, and we don't have logs from when it thinks time was synchronized.~~ Sled-agent believes time was synchronized prior to the above snippet:

```
/staff/core/omicron-4087 $ grep 'Time is sync' oxide-sled-agent\:default.log.1694562293  | looker
20:39:38.278Z INFO SledAgent (ServiceManager): Time is synchronized
    file = sled-agent/src/services.rs:567
```

**Comment by iliana** [comment]
What we think happened here is the boundary NTP zones considered themselves authoritative without having upstream connectivity, and responded to other NTP zones with a 1980-era timestamp. The telling factor here is the stratum 11 lines in tracking.log, which imply the boundary NTP service was serving stratum 10:

https://github.com/oxidecomputer/omicron/blob/23eee1a9afe10e1e63f4cc1bbdf4a7014515a8e4/smf/ntp/etc/inet/chrony.conf.boundary#L19-L22

<details><summary>`head -n 100 /var/log/chrony/tracking.log` in BRM42220017's NTP zone</summary>

```
===================================================================================================================================
   Date (UTC) Time     IP Address   St   Freq ppm   Skew ppm     Offset L Co  Offset sd Rem. corr. Root delay Root disp. Max. error
===================================================================================================================================
1986-12-28 00:04:17 0.0.0.0          0     -0.000 1000000.000  0.000e+00 ?  0  0.000e+00  0.000e+00  1.000e+00  1.000e+00  1.500e+00
1986-12-28 00:04:32 fd00:1122:3344:104::e 11     -1.937    236.343 -2.138e+01 N  1  2.381e-07  0.000e+00  9.770e-05  4.842e-03  1.500e+00
1986-12-28 00:04:54 fd00:1122:3344:104::e 11     -1.937    236.343 -2.290e-03 N  1  5.884e-04  2.137e+01  1.303e-04  1.855e-03  2.138e+01
1986-12-28 00:04:55 fd00:1122:3344:104::e 11     -1.937    236.343 -8.174e-04 N  1  6.988e-04  2.137e+01  1.844e-04  8.358e-04  2.138e+01
1986-12-28 00:04:56 fd00:1122:3344:104::e 11     -1.937    236.343  2.566e-04 N  1  4.807e-04  2.137e+01  1.146e-04  5.762e-04  2.138e+01
1986-12-28 00:04:57 fd00:1122:3344:104::e 11    -71.712    514.839 -4.534e-05 N  1  4.083e-04  2.137e+01  9.389e-05  1.155e-03  2.137e+01
1986-12-28 00:04:58 fd00:1122:3344:104::e 11   -270.128    790.411 -6.849e-05 N  1  3.749e-04  2.137e+01  1.124e-04  4.397e-04  2.137e+01
1986-12-28 00:04:59 fd00:1122:3344:104::e 11    -33.751    216.989  6.270e-04 N  1  5.756e-05  2.136e+01  1.526e-04  1.205e-04  2.137e+01
1986-12-28 00:05:05 fd00:1122:3344:104::e  4    -33.751    216.989 -1.158e+09 N  1  2.977e-03  2.135e+01  6.232e-02  3.393e+06  2.136e+01
2023-09-12 20:39:39 fd00:1122:3344:104::e  4    -33.751    216.989 -4.365e-01 N  1  3.490e-05 -1.849e-07  6.232e-02  4.810e-01  1.158e+09
2023-09-12 20:39:40 fd00:1122:3344:104::e  4    -33.751    216.989 -2.606e-03 N  1  1.185e-04  4.338e-01  6.232e-02  4.680e-01  1.132e+00
2023-09-12 20:39:41 fd00:1122:3344:104::e  4    -33.751    216.989 -5.917e-04 N  1  5.313e-04  4.337e-01  6.232e-02  4.708e-01  9.385e-01
2023-09-12 20:39:42 fd00:1122:3344:104::e  4    -33.751    216.989  1.948e-04 N  1  4.825e-04  4.316e-01  6.232e-02  4.713e-01  9.380e-01
2023-09-12 20:39:43 fd00:1122:3344:104::e  4    -33.751    216.989 -3.542e-04 N  1  5.661e-04  4.286e-01  6.232e-02  4.724e-01  9.350e-01
2023-09-12 20:39:44 fd00:1122:3344:104::e  4    -33.751    216.989 -1.507e-04 N  1  6.236e-04  4.262e-01  6.232e-02  4.734e-01  9.335e-01
2023-09-12 20:39:45 fd00:1122:3344:104::e 11    -54.600     36.231  7.290e-04 N  1  1.095e-05  4.237e-01  2.177e-04  1.473e-05  9.317e-01
2023-09-12 20:39:46 fd00:1122:3344:104::e 11    -55.315     21.146 -2.841e-06 N  1  1.018e-05  4.202e-01  2.403e-04  1.528e-05  4.231e-01
2023-09-12 20:39:47 fd00:1122:3344:104::e 11    -55.171     13.377  6.641e-07 N  1  8.949e-06  4.174e-01  2.372e-04  1.882e-05  4.203e-01
2023-09-12 20:39:48 fd00:1122:3344:104::e 11    -56.157      8.764 -5.939e-06 N  1  6.926e-06  4.147e-01  1.916e-04  1.241e-05  4.176e-01
2023-09-12 20:39:49 fd00:1122:3344:104::e 11    -55.024      6.559  8.625e-06 N  1  5.673e-06  4.120e-01  1.706e-04  6.959e-06  4.148e-01
2023-09-12 20:39:50 fd00:1122:3344:104::e 11    -55.253      5.606 -2.901e-06 N  1  6.254e-06  4.092e-01  1.706e-04  1.795e-05  4.121e-01
2023-09-12 20:39:51 fd00:1122:3344:104::e 11    -55.712      4.428 -3.652e-06 N  1  5.223e-06  4.065e-01  1.734e-04  9.056e-06  4.094e-01
2023-09-12 20:39:52 fd00:1122:3344:104::e 11    -56.050      3.678 -2.627e-06 N  1  4.857e-06  4.038e-01  1.734e-04  1.153e-05  4.066e-01
2023-09-12 20:39:53 fd00:1122:3344:104::e 11    -55.994      3.030  1.191e-06 N  1  4.598e-06  4.011e-01  1.734e-04  1.173e-05  4.039e-01
2023-09-12 20:39:54 fd00:1122:3344:104::e 11    -55.622      2.797  4.195e-06 N  1  4.333e-06  3.983e-01  1.739e-04  7.205e-06  4.012e-01
2023-09-12 20:39:55 fd00:1122:3344:104::e 11    -56.287      5.067  2.326e-05 N  1  7.427e-06  3.956e-01  1.739e-04  1.826e-05  3.984e-01
2023-09-12 20:39:56 fd00:1122:3344:104::e 11    -55.667      7.019  1.058e-05 N  1  8.682e-06  3.928e-01  1.739e-04  2.537e-05  3.957e-01
2023-09-12 20:39:57 fd00:1122:3344:104::e 11    -55.090      6.389  6.849e-07 N  1  7.970e-06  3.901e-01  2.026e-04  8.571e-06  3.929e-01
2023-09-12 20:39:58 fd00:1122:3344:104::e 11    -52.195     10.690  2.320e-05 N  1  9.168e-06  3.874e-01  2.026e-04  3.095e-05  3.902e-01
2023-09-12 20:39:59 0.0.0.0          0    -52.195     10.690  0.000e+00 ?  0  0.000e+00  3.855e-01  2.026e-04  4.050e-05  3.875e-01
2023-09-12 20:40:02 fd00:1122:3344:104::e 11    -51.886      3.506 -3.871e-05 N  1  5.181e-06  3.773e-01  1.739e-04  3.162e-05  3.856e-01
2023-09-12 20:40:02 fd00:1122:3344:104::e 11    -52.334      3.366 -5.130e-06 N  1  5.548e-06  3.764e-01  1.739e-04  3.338e-05  3.774e-01
===================================================================================================================================
   Date (UTC) Time     IP Address   St   Freq ppm   Skew ppm     Offset L Co  Offset sd Rem. corr. Root delay Root disp. Max. error
===================================================================================================================================
2023-09-12 20:40:03 fd00:1122:3344:104::e 11    -53.326      4.515 -8.966e-06 N  1  5.721e-06  3.737e-01  1.934e-04  1.138e-05  3.765e-01
2023-09-12 20:40:04 fd00:1122:3344:104::e 11    -54.117      4.090 -4.592e-06 N  1  5.717e-06  3.709e-01  1.934e-04  1.373e-05  3.738e-01
2023-09-12 20:40:05 fd00:1122:3344:104::e 11    -54.690      3.398 -4.895e-06 N  1  5.250e-06  3.682e-01  1.939e-04  8.103e-06  3.711e-01
2023-09-12 20:40:07 fd00:1122:3344:104::e 11    -54.455      2.931  3.916e-06 N  1  5.798e-06  3.628e-01  1.939e-04  1.418e-05  3.684e-01
2023-09-12 20:40:08 0.0.0.0          0    -54.455      2.931  0.000e+00 ?  0  0.000e+00  3.609e-01  1.939e-04  1.701e-05  3.629e-01
2023-09-12 20:42:16 fd00:1122:3344:102::e 11    -54.455      2.931  7.329e-01 N  1  1.164e-01  1.421e-02  2.319e-04  1.267e-01  3.615e-01
2023-09-12 20:42:16 fd00:1122:3344:102::e 11    -54.455      2.931 -2.286e-01 N  1  6.688e-02 -7.159e-01  1.432e-04  7.104e-02  8.484e-01
2023-09-12 20:42:17 fd00:1122:3344:102::e 11    -54.455      2.931 -4.391e-02 N  1  3.282e-02 -4.845e-01  1.115e-04  3.808e-02  5.890e-01
2023-09-12 20:42:19 fd00:1122:3344:102::e 11    -54.455      2.931 -2.434e-02 N  1  2.406e-02 -4.373e-01  1.026e-04  4.596e-02  5.189e-01
2023-09-12 20:42:20 fd00:1122:3344:102::e 11    -54.455      2.931 -5.830e-03 N  1  2.560e-02 -4.102e-01  1.110e-04  5.038e-02  4.954e-01
2023-09-12 20:42:21 fd00:1122:3344:102::e 11    -54.451      3.555  3.734e-02 N  1  2.271e-04 -4.016e-01  1.273e-04  2.542e-04  4.847e-01
2023-09-12 20:42:22 fd00:1122:3344:102::e 11    -40.635     29.256 -1.012e-05 N  1  3.684e-06 -4.362e-01  1.170e-04  1.558e-05  4.394e-01
2023-09-12 20:42:23 fd00:1122:3344:102::e 11     -2.292     18.847  3.568e-05 N  1  5.684e-06 -4.335e-01  1.359e-04  1.577e-05  4.363e-01
2023-09-12 20:42:24 fd00:1122:3344:102::e 11      0.440     11.503  8.542e-06 N  1  7.635e-06 -4.307e-01  1.466e-04  1.907e-05  4.336e-01
2023-09-12 20:42:25 fd00:1122:3344:102::e 11      1.275      8.031  4.258e-06 N  1  7.046e-06 -4.280e-01  1.324e-04  1.255e-05  4.308e-01
2023-09-12 20:42:26 fd00:1122:3344:102::e 11      0.413      5.645 -7.104e-06 N  1  5.827e-06 -4.252e-01  1.143e-04  6.094e-06  4.281e-01
2023-09-12 20:42:27 fd00:1122:3344:102::e 11     -0.861      8.570 -1.100e-05 N  1  5.837e-06 -4.225e-01  9.177e-05  1.402e-05  4.253e-01
2023-09-12 20:42:28 fd00:1122:3344:102::e 11     -1.648     14.347 -7.012e-06 N  1  7.247e-06 -4.197e-01  9.774e-05  2.422e-05  4.225e-01
2023-09-12 20:42:29 fd00:1122:3344:102::e 11     -6.428     16.108 -5.379e-06 N  1  3.469e-06 -4.170e-01  8.035e-05  6.099e-06  4.198e-01
2023-09-12 20:42:30 fd00:1122:3344:102::e 11     -4.656     15.446  7.401e-06 N  1  3.604e-06 -4.142e-01  9.132e-05  5.653e-06  4.170e-01
2023-09-12 20:42:31 fd00:1122:3344:102::e 11     -2.038     11.768  6.902e-06 N  1  3.170e-06 -4.115e-01  1.024e-04  8.860e-06  4.143e-01
2023-09-12 20:42:32 fd00:1122:3344:102::e 11     -1.143      8.358  2.899e-06 N  1  3.865e-06 -4.087e-01  1.024e-04  1.254e-05  4.116e-01
2023-09-12 20:42:33 fd00:1122:3344:102::e 11     -1.362      5.231 -1.478e-06 N  1  3.460e-06 -4.060e-01  1.049e-04  3.599e-06  4.088e-01
2023-09-12 20:42:34 fd00:1122:3344:102::e 11     -1.593      3.373 -1.461e-06 N  1  2.793e-06 -4.032e-01  9.443e-05  5.905e-06  4.060e-01
2023-09-12 20:42:35 fd00:1122:3344:102::e 11     -1.364      3.204  2.116e-06 N  1  3.426e-06 -4.005e-01  9.443e-05  8.666e-06  4.033e-01
2023-09-12 20:42:36 fd00:1122:3344:102::e 11     -1.448      2.546 -9.842e-07 N  1  3.210e-06 -3.977e-01  9.443e-05  9.839e-06  4.005e-01
2023-09-12 20:42:37 fd00:1122:3344:102::e 11     -1.462      2.089  3.533e-08 N  1  3.128e-06 -3.950e-01  9.443e-05  1.121e-05  3.978e-01
2023-09-12 20:42:38 fd00:1122:3344:102::e 11     -2.236      3.987 -1.075e-05 N  1  4.324e-06 -3.923e-01  9.548e-05  5.682e-06  3.951e-01
2023-09-12 20:42:39 fd00:1122:3344:102::e 11     -3.180      3.525 -1.784e-06 N  1  3.580e-06 -3.895e-01  9.548e-05  8.294e-06  3.923e-01
2023-09-12 20:42:40 fd00:1122:3344:102::e 11     -3.177      3.110  1.332e-06 N  1  4.343e-06 -3.868e-01  9.548e-05  1.452e-05  3.896e-01
2023-09-12 20:42:41 fd00:1122:3344:102::e 11     -3.159      2.136  2.506e-07 N  1  3.037e-06 -3.840e-01  9.008e-05  4.511e-06  3.868e-01
2023-09-12 20:42:42 fd00:1122:3344:102::e 11     -3.280      2.147 -1.362e-06 N  1  3.603e-06 -3.813e-01  9.008e-05  7.983e-06  3.841e-01
===================================================================================================================================
   Date (UTC) Time     IP Address   St   Freq ppm   Skew ppm     Offset L Co  Offset sd Rem. corr. Root delay Root disp. Max. error
===================================================================================================================================
2023-09-12 20:42:43 fd00:1122:3344:102::e 11     -3.219      2.042  1.222e-06 N  1  3.882e-06 -3.785e-01  9.008e-05  8.328e-06  3.813e-01
2023-09-12 20:42:44 fd00:1122:3344:102::e 11     -3.121      1.975  1.281e-06 N  1  4.154e-06 -3.758e-01  9.008e-05  1.171e-05  3.786e-01
2023-09-12 20:42:45 fd00:1122:3344:102::e 11     -3.193      1.799 -1.115e-06 N  1  4.098e-06 -3.730e-01  9.008e-05  1.288e-05  3.758e-01
2023-09-12 20:42:46 fd00:1122:3344:102::e 11     -3.164      1.683  8.057e-07 N  1  4.281e-06 -3.703e-01  9.008e-05  1.336e-05  3.731e-01
2023-09-12 20:42:47 fd00:1122:3344:102::e 11     -3.147      1.369  7.128e-09 N  1  3.633e-06 -3.675e-01  1.023e-04  4.501e-06  3.704e-01
2023-09-12 20:42:49 fd00:1122:3344:102::e 11     -3.086      1.316  1.241e-06 N  1  4.101e-06 -3.620e-01  1.023e-04  7.988e-06  3.676e-01
2023-09-12 20:42:51 fd00:1122:3344:102::e 11     -3.185      1.172 -2.257e-06 N  1  3.999e-06 -3.565e-01  1.023e-04  9.323e-06  3.621e-01
2023-09-12 20:42:53 fd00:1122:3344:102::e 11     -3.114      1.130  2.021e-06 N  1  4.465e-06 -3.509e-01  1.023e-04  1.311e-05  3.565e-01
2023-09-12 20:42:55 fd00:1122:3344:102::e 11     -3.391      1.105 -5.974e-06 N  1  4.013e-06 -3.455e-01  7.674e-05  6.115e-06  3.510e-01
2023-09-12 20:42:57 fd00:1122:3344:102::e 11     -3.379      0.940  1.504e-06 N  1  4.247e-06 -3.400e-01  7.674e-05  7.647e-06  3.455e-01
2023-09-12 20:43:01 fd00:1122:3344:102::e 11     -3.400      0.841 -5.899e-07 N  1  4.528e-06 -3.291e-01  7.674e-05  9.492e-06  3.401e-01
2023-09-12 20:43:05 fd00:1122:3344:102::e 11     -3.301      0.841  3.632e-06 N  1  5.085e-06 -3.181e-01  7.674e-05  1.478e-05  3.292e-01
2023-09-12 20:43:09 fd00:1122:3344:102::e 11     -3.223      0.768  2.214e-06 N  1  5.324e-06 -3.071e-01  7.674e-05  1.626e-05  3.182e-01
2023-09-12 20:43:13 fd00:1122:3344:102::e 11     -3.152      0.685  2.447e-06 N  1  5.316e-06 -2.962e-01  7.674e-05  1.821e-05  3.072e-01
2023-09-12 20:43:22 fd00:1122:3344:102::e  4     -3.153      0.958 -2.933e-03 N  1  1.328e-03 -2.741e-01  1.346e-04  7.216e-03  2.963e-01
2023-09-12 20:43:23 fd00:1122:3344:102::e  4     -3.155      1.714 -1.797e-03 N  1  9.286e-04 -2.684e-01  1.346e-04  6.318e-03  2.786e-01
2023-09-12 20:43:24 fd00:1122:3344:102::e  4     -3.164      3.617 -7.698e-04 N  1  6.505e-04 -2.638e-01  1.346e-04  6.955e-03  2.733e-01
2023-09-12 20:43:25 fd00:1122:3344:102::e  4     -3.246      9.624 -3.530e-04 N  1  3.633e-04 -2.603e-01  1.346e-04  5.725e-03  2.705e-01
2023-09-12 20:43:26 fd00:1122:3344:102::e  4     -3.679     24.613 -4.545e-04 N  1  3.298e-04 -2.572e-01  1.346e-04  7.483e-03  2.662e-01
2023-09-12 20:43:27 fd00:1122:3344:102::e  4    -70.087     46.557  5.747e-04 N  1  5.195e-06 -2.540e-01  6.241e-02  1.888e-03  2.648e-01
2023-09-12 20:43:28 fd00:1122:3344:102::e  4    -83.987     46.811 -3.150e-05 N  1  1.165e-05 -2.518e-01  6.237e-02  2.001e-03  2.877e-01
2023-09-12 20:43:29 fd00:1122:3344:102::e  4    -91.487     40.234 -1.637e-05 N  1  1.117e-05 -2.491e-01  6.237e-02  2.035e-03  2.851e-01
2023-09-12 20:43:30 fd00:1122:3344:102::e  4    -93.491     30.429 -2.275e-06 N  1  1.042e-05 -2.463e-01  6.237e-02  2.031e-03  2.823e-01
2023-09-12 20:43:31 fd00:1122:3344:102::e  4    -91.555     22.938  1.112e-05 N  1  1.207e-05 -2.435e-01  6.238e-02  2.015e-03  2.795e-01
2023-09-12 20:43:32 fd00:1122:3344:102::e  4    -90.185     29.799  8.367e-06 N  1  1.418e-05 -2.408e-01  6.237e-02  2.106e-03  2.768e-01
2023-09-12 20:43:33 fd00:1122:3344:102::e  4    -88.681     21.256  4.330e-06 N  1  1.241e-05 -2.380e-01  6.237e-02  2.056e-03  2.741e-01
2023-09-12 20:43:34 fd00:1122:3344:102::e  4    -85.511     19.239  9.330e-06 N  1  5.921e-06 -2.353e-01  6.241e-02  2.079e-03  2.713e-01
```

</details>

We could most easily work around this by ensuring we don't consider the host in sync until the stratum is < 10. Maybe that's not good enough; you can see a five second period where we are stratum 4, but then quickly return back to stratum 11 as the upstream realizes it has jumped and needs to resynchronize. (By the time we get back down to stratum 4 though we've only lost true time by -2.933 ms, which is well within what both us and CRDB cares about.) We could also set up an epoch before which time cannot be to consider ourselves in sync, which might be a good additional defense?

Ideally, though, the boundary NTP zones would not start until we've confirmed external connectivity. Not sure where in rack setup this check would/could take place.

**Comment by davepacheco** [comment]
@iliana was this issue resolved by #4119?

**Comment by iliana** [comment]
Seems likely!
